world_model_ed.py: center_point.vector.x(): Point has no attribute 'vector'

File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 381, in _publish_marker
    
marker.pose.position.x = center_point.vector.x()
should be 
marker.pose.position.x = center_point.x

[INFO] [1491936113.471354]: State machine transitioning 'WAIT_TO_FOLLOW':'success'-->'FOLLOW_OPERATOR'
/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/torso.py:132: Warning: Please use wait_for_motion_done instead
  warnings.warn("Please use wait_for_motion_done instead", Warning)
x: -0.148067480855
y: 2.37502062155
z: 1.0
[ERROR] [1491936113.497058]: InvalidUserCodeError: Could not execute state 'FOLLOW_OPERATOR' of type '<robot_smach_states.navigation.follow_operator.FollowOperator object at 0x7ff8b6b8f750>': Traceback (most recent call last):
  File "/home/amigo/ros/kinetic/system/src/smach/src/smach/state_machine.py", line 247, in _update_once
    self._remappings[self._current_label]))
  File "/home/amigo/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/follow_operator.py", line 604, in execute
    if not self._register_operator():
  File "/home/amigo/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/follow_operator.py", line 166, in _register_operator
    operator = self._robot.ed.get_closest_possible_person_entity(radius=1, center_point=msg_constructors.PointStamped(x=1.5, y=0, z=1, frame_id="/%s/base_link"%self._robot.robot_name))
  File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 251, in get_closest_possible_person_entity
    entities = self.get_entities(type="", center_point=center_point, radius=radius)
  File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 83, in get_entities
    self._publish_marker(center_point, radius)
  File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 381, in _publish_marker
    marker.pose.position.x = center_point.vector.x()
AttributeError: 'Point' object has no attribute 'vector'

[ERROR] [1491936113.497448]: Robot exited with <class 'smach.exceptions.InvalidUserCodeError'>,Could not execute state 'FOLLOW_OPERATOR' of type '<robot_smach_states.navigation.follow_operator.FollowOperator object at 0x7ff8b6b8f750>': Traceback (most recent call last):
  File "/home/amigo/ros/kinetic/system/src/smach/src/smach/state_machine.py", line 247, in _update_once
    self._remappings[self._current_label]))
  File "/home/amigo/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/follow_operator.py", line 604, in execute
    if not self._register_operator():
  File "/home/amigo/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/follow_operator.py", line 166, in _register_operator
    operator = self._robot.ed.get_closest_possible_person_entity(radius=1, center_point=msg_constructors.PointStamped(x=1.5, y=0, z=1, frame_id="/%s/base_link"%self._robot.robot_name))
  File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 251, in get_closest_possible_person_entity
    entities = self.get_entities(type="", center_point=center_point, radius=radius)
  File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 83, in get_entities
    self._publish_marker(center_point, radius)
  File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 381, in _publish_marker
    marker.pose.position.x = center_point.vector.x()
AttributeError: 'Point' object has no attribute 'vector'
,<traceback object at 0x7ff8b6b4ed88>
[INFO] [1491936113.497773]: Torso cancelling all goals on close
[INFO] [1491936113.498238]: left arm cancelling all goals on all arm-related ACs on close
[INFO] [1491936113.498882]: right arm cancelling all goals on all arm-related ACs on close
Traceback (most recent call last):
  File "/home/amigo/ros/kinetic/system/src/challenge_help_me_carry/src/challenge_help_me_carry.py", line 249, in <module>
    states.util.startup(setup_statemachine, challenge_name="help_me_carry")
  File "/home/amigo/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/util/startup.py", line 78, in startup
    outcome = executioner.execute()
  File "/home/amigo/ros/kinetic/system/src/smach/src/smach/state_machine.py", line 359, in execute
    container_outcome = self._update_once()
  File "/home/amigo/ros/kinetic/system/src/smach/src/smach/state_machine.py", line 254, in _update_once
    + traceback.format_exc())
smach.exceptions.InvalidUserCodeError: Could not execute state 'FOLLOW_OPERATOR' of type '<robot_smach_states.navigation.follow_operator.FollowOperator object at 0x7ff8b6b8f750>': Traceback (most recent call last):
  File "/home/amigo/ros/kinetic/system/src/smach/src/smach/state_machine.py", line 247, in _update_once
    self._remappings[self._current_label]))
  File "/home/amigo/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/follow_operator.py", line 604, in execute
    if not self._register_operator():
  File "/home/amigo/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/navigation/follow_operator.py", line 166, in _register_operator
    operator = self._robot.ed.get_closest_possible_person_entity(radius=1, center_point=msg_constructors.PointStamped(x=1.5, y=0, z=1, frame_id="/%s/base_link"%self._robot.robot_name))
  File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 251, in get_closest_possible_person_entity
    entities = self.get_entities(type="", center_point=center_point, radius=radius)
  File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 83, in get_entities
    self._publish_marker(center_point, radius)
  File "/home/amigo/ros/kinetic/system/src/robot_skills/src/robot_skills/world_model_ed.py", line 381, in _publish_marker
    marker.pose.position.x = center_point.vector.x()
AttributeError: 'Point' object has no attribute 'vector'

^C[state_machine-7] killing on exit
[amigo/object_recognition-6] killing on exit
[amigo/face_recognition-5] killing on exit
[amigo/ed-3] killing on exit
[amigo/global_planner-2] killing on exit
[amigo/local_planner-1] killing on exit
[state_machine-7] escalating to SIGTERM
[state_machine-7] escalating to SIGKILL
Shutdown errors:
 * process[state_machine-7, pid 7524]: required SIGKILL. May still be running.
shutting down processing monitor...
... shutting down processing monitor complete
done

--- a/robot_skills/src/robot_skills/world_model_ed.py
+++ b/robot_skills/src/robot_skills/world_model_ed.py
@@ -244,7 +244,7 @@ class ED(RobotPart):
 
     # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
-    def get_closest_possible_person_entity(self, type="", center_point=Point(), radius=0, room = ""):
+    def get_closest_possible_person_entity(self, type="", center_point=VectorStamped(), radius=0, room = ""):
         if isinstance(center_point, PointStamped):
             center_point = self._transform_center_point_to_map(center_point)
 

--- a/robot_smach_states/src/robot_smach_states/navigation/follow_operator.py
+++ b/robot_smach_states/src/robot_smach_states/navigation/follow_operator.py
@@ -163,7 +163,7 @@ class FollowOperator(smach.State):
                     self._robot.speech.speak("Something is wrong with my ears, please take a look!")
                     return False
             else:
-                operator = self._robot.ed.get_closest_possible_person_entity(radius=1, center_point=msg_constructors.PointStamped(x=1.5, y=0, z=1, frame_id="/%s/base_link"%self._robot.robot_name))
+                operator = self._robot.ed.get_closest_possible_person_entity(radius=1, center_point=VectorStamped(x=1.5, y=0, z=1, frame_id="/%s/base_link"%self._robot.robot_name))
                 if not operator:
                     rospy.sleep(1)
 
@@ -211,7 +211,7 @@ class FollowOperator(smach.State):
         # This only happens when the operator was just registered, and never tracked
         print "Operator already lost. Getting closest possible person entity at 1.5 m in front, radius = 1"
         self._operator = self._robot.ed.get_closest_possible_person_entity(radius=1,
-                                                                                center_point=msg_constructors.PointStamped(
+                                                                                center_point=VectorStamped(
                                                                                     x=1.5, y=0, z=1,
                                                                                     frame_id="/%s/base_link" % self._robot.robot_name))
         if self._operator:
@@ -219,7 +219,7 @@ class FollowOperator(smach.State):
         else:
             print "Operator still lost. Getting closest possible laser entity at 1.5 m in front, radius = 1"
             self._operator = self._robot.ed.get_closest_laser_entity(radius=1,
-                                                                          center_point=msg_constructors.PointStamped(
+                                                                          center_point=VectorStamped(
                                                                               x=1.5, y=0, z=1,
                                                                               frame_id="/%s/base_link" % self._robot.robot_name)
                                                                           )

